### PR TITLE
feat: renameProject

### PR DESCRIPTION
**Pull request by lloydAffinidi**
https://user-images.githubusercontent.com/118436059/212124009-8c3beeae-0b67-451d-ae1c-8d0da5a18140.mov

Successful rename: 
Screenshot 2023-01-13 at 16 41 27

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -236,6 +237,18 @@ export async function activateInternal(context: ExtensionContext) {
     },
   )
 
+  commands.registerCommand(
+    'affinidiExplorer.renameProject',
+    (element: BasicTreeItemWithProject) => {
+      telemetryHelpers.trackCommand('affinidiExplorer.renameProject', {
+        projectId: element.projectId,
+        resource: element.contextValue,
```
Comment (carlos-affinidi) [inline src/extension.ts:245]: `resource` not needed. We know its a project.

**Review by carlos-affinidi — CHANGES_REQUESTED** [review]

File: src/extension.ts (hunk 1 above)
Comment (lloydAffinidi) [reply src/extension.ts]: removed

File: package.json [hunk 2]
```diff
@@ -377,6 +387,11 @@
           "when": "view == affinidiExplorer && viewItem == inactiveProject || view == affinidiExplorer && viewItem == project || view == affinidiExplorer && viewItem == schema || view == affinidiExplorer && viewItem == issuance",
           "group": "inline"
         },
+        {
+          "command": "affinidiExplorer.renameProject",
+          "when": "view == affinidiExplorer && viewItem == inactiveProject || view == affinidiExplorer && viewItem == project || view == affinidiExplorer && viewItem == schema || view == affinidiExplorer && viewItem == issuance",
```
Comment (vitalyrudenko-affinidi) [inline package.json:392]: Not sure why schema and issuance are here?

**Review by carlos-affinidi — APPROVED** [review]

File: src/extension.ts [hunk 3]
```diff
@@ -236,6 +237,17 @@ export async function activateInternal(context: ExtensionContext) {
...
   )
 
+  commands.registerCommand(
+    'affinidiExplorer.renameProject',
+    async (element: BasicTreeItemWithProject) => {
+      telemetryHelpers.trackCommand('affinidiExplorer.renameProject', {
+        projectId: element.projectId,
+      })
+
+      renameProjectService.activateRenameProject(element)
```
Comment (vitalyrudenko-affinidi) [inline src/extension.ts:247]:
```suggestion
      await iamHelpers.renameProject({ projectId: element.projectId })
```

File: src/features/iam/renameProjects.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+  if (name) {
+    await renameProject({ projectId, name })
+
+    iamState.clear()
+    ext.explorerTree.refresh()
+  }
+}
+
+export const renameProjectService = { activateRenameProject, renameProject }
```
Comment (vitalyrudenko-affinidi) [inline src/features/iam/renameProjects.ts:51]: I appreciate the effort of splitting this into two functions, and I think we can simplify it a little bit more:
```ts
const renameProject = async (input: { projectId: string }) => {
  const { projectId } = input

  const project = iamState.getProject(projectId)
  const name = await window.showInputBox({
    prompt: `Enter the new name for the project "${project.name}":`,
  })
  if (!name) return

  try {
    const consoleAuthToken = await authHelper.getConsoleAuthToken()
    const data = await window.withProgress(
      {
        location: ProgressLocation.Notification,
        title: projectMessage.renameProject,
      },
      async () => {
        return iamClient.patchProject({ projectId, name: { name } }, { consoleAuthToken })
      },
    )

    window.showInformationMessage(projectMessage.projectRenamed)
    ext.outputChannel.appendLine(JSON.stringify(data, null, 2))
    ext.outputChannel.show()

    iamState.clear()
    ext.explorerTree.refresh()
  } catch (error: unknown) {
    notifyError(error, projectMessage.projectRenameFailed)
  }
}
```

And I think we need to move this function to `iamHelpers.ts`

File: src/features/iam/renameProjects.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  const { name } = input
+
+  try {
+    const consoleAuthToken = await authHelper.getConsoleAuthToken()
+    const data = await window.withProgress(
+      {
+        location: ProgressLocation.Notification,
+        title: projectMessage.renameProject,
+      },
+      async () => {
```
Comment (OleksiiO-affinidi) [inline src/features/iam/renameProjects.ts:21]: no need in `async` here

**Pull request merged**